dev/PipsControl/PipsControl.cpp

```diff
@@ -0,0 +1,369 @@
+    if (const auto scrollViewer = m_verticalPipsScrollViewer.get()) {
+        m_verticalPipsScrollViewer.get().MaxHeight(scrollViewerHeight);
+    }
+}
+
+
+
+void PipsControl::UpdateVerticalPips(const int numberOfPages, const int maxDisplayedPages) {
+
+    auto pipsListSize = static_cast<int>(m_verticalPipsElements.Size());
+    auto const selectedIndex = SelectedPageIndex();
+
+    if (numberOfPages < pipsListSize) {
+        m_verticalPipsElements.Clear();
+        pipsListSize = 0;
+    }
+
+    auto const endIndex = std::min(numberOfPages, selectedIndex + maxDisplayedPages);
+    for (int i = pipsListSize; i < endIndex; i++) {
+        m_verticalPipsElements.Append(box_value(i + 1));
+    }
+
+    if (maxDisplayedPages != m_lastMaxDisplayedPages) {
+        SetVerticalPipsSVMaxSize();
+    }
+    MovePipIdentifierToElement(SelectedPageIndex());
+
+}
+
+void PipsControl::OnRootGridKeyDown(const winrt::IInspectable& sender, const winrt::KeyRoutedEventArgs& args) {
+    if (args.Key() == winrt::VirtualKey::Left || args.Key() == winrt::VirtualKey::Up)
+    {
+        winrt::FocusManager::TryMoveFocus(winrt::FocusNavigationDirection::Up);
+    }
+    else if (args.Key() == winrt::VirtualKey::Right || args.Key() == winrt::VirtualKey::Down)
+    {
+        winrt::FocusManager::TryMoveFocus(winrt::FocusNavigationDirection::Down);
+    }
+}
+
+
+void PipsControl::OnPreviousButtonClicked(const IInspectable& sender, const winrt::RoutedEventArgs& e)
+{
+    // In this method, SelectedPageIndex is always greater than 1.
+       SelectedPageIndex(SelectedPageIndex() - 1);
+}
+
+void PipsControl::OnNextButtonClicked(const IInspectable& sender, const winrt::RoutedEventArgs& e)
+{
+    // In this method, SelectedPageIndex is always less than maximum.
+    SelectedPageIndex(SelectedPageIndex() + 1);
+}
+
+void PipsControl::OnPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
+{
+    winrt::IDependencyProperty property = args.Property();
+    if (this->Template() != nullptr)
+    {
+        if (property == NumberOfPagesProperty())
+        {
+            OnNumberOfPagesChanged(winrt::unbox_value<int>(args.OldValue()));
+        }
+        else if (property == SelectedPageIndexProperty())
+        {
+            OnSelectedPageIndexChange(winrt::unbox_value<int>(args.OldValue()));
+        }
+        else if (property == MaxDisplayedPagesProperty()) {
+            OnMaxDisplayedPagesChanged(winrt::unbox_value<int>(args.OldValue()));
+        }
+        else if (property == PreviousButtonVisibilityProperty())
+        {
+            OnNavigationButtonVisibilityChanged(PreviousButtonVisibility(), c_previousPageButtonCollapsedVisualState);
+        }
+        else if (property == NextButtonVisibilityProperty())
+        {
+            OnNavigationButtonVisibilityChanged(NextButtonVisibility(), c_nextPageButtonCollapsedVisualState);
+        }
+        // TODO: Add button styles listeners
```

Comment:
>Add [](start = 17, length = 3)

I don't think you'll need to add listeners, they should just work with template bindings in the tempalte.